src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1106,6 +1115,116 @@ CreateSchemaDDLCommand(Oid schemaId)
 }
 
 
+/*
+ * GrantOnSchemaDDLCommands creates a list of ddl command for replicating the permissions
+ * of roles on schemas.
+ */
+List *
+GrantOnSchemaDDLCommands(Oid schemaOid)
+{
+	HeapTuple schemaTuple = SearchSysCache1(NAMESPACEOID, ObjectIdGetDatum(schemaOid));
+	bool isNull = true;
+	Datum aclDatum = SysCacheGetAttr(NAMESPACEOID, schemaTuple, Anum_pg_namespace_nspacl,
+									 &isNull);
+	if (isNull)
+	{
+		ReleaseSysCache(schemaTuple);
+		return NIL;
+	}
+	Acl *acl = DatumGetAclPCopy(aclDatum);
+	AclItem *aclDat = ACL_DAT(acl);
+	int aclNum = ACL_NUM(acl);
+	List *commands = NIL;
+	Oid currentUser = GetUserId();
+
+	ReleaseSysCache(schemaTuple);
+
+	for (int i = 0; i < aclNum; i++)
+	{
+		bool withSetRole = (currentUser != aclDat[i].ai_grantor);
```

Comment:
This check is incorrect here :(. Due to this I was able to get grants created by the wrong user on a worker.

The scenario
 - SCHEMA foo
 - super user grants user_1
 - user_1 grants user_2
 - user_1 creates table
 - user_1 distributes table (causing the schema to propagate)

Now the grant for user_2 is granted by the superuser on the workers

```sql
CREATE USER user_1;
CREATE USER user_2;
SELECT run_command_on_workers($$CREATE USER user_1;$$);
SELECT run_command_on_workers($$CREATE USER user_2;$$);

CREATE SCHEMA foo;
GRANT ALL ON SCHEMA foo TO user_1 WITH GRANT OPTION;
SET ROLE user_1;
GRANT ALL ON SCHEMA foo TO user_2 WITH GRANT OPTION;
CREATE TABLE foo.bar (a int PRIMARY KEY, b int);
SELECT create_distributed_table('foo.bar','a');
```

now the coordinator has the following ACL:
```sql
SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('foo') ORDER BY nspname;
┌─────────┬────────────────────────────────────────────────────────────────┐
│ nspname │                             nspacl                             │
├─────────┼────────────────────────────────────────────────────────────────┤
│ foo     │ {nilsdijk=UC/nilsdijk,user_1=U*C*/nilsdijk,user_2=U*C*/user_1} │
└─────────┴────────────────────────────────────────────────────────────────┘
```

while the worker shows:
```sql
SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('foo') ORDER BY nspname;
┌─────────┬──────────────────────────────────────────────────────────────────┐
│ nspname │                              nspacl                              │
├─────────┼──────────────────────────────────────────────────────────────────┤
│ foo     │ {nilsdijk=UC/nilsdijk,user_1=U*C*/nilsdijk,user_2=U*C*/nilsdijk} │
└─────────┴──────────────────────────────────────────────────────────────────┘
```

The user creating the DDL commands that will be propagated is not always the same user that will send the commands.

Reply:
thinking more about it, I think we can always include the `SET ROLE` commands in this codepath